src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultModelBindingContext.cs

```diff
@@ -104,18 +108,25 @@ public DefaultModelBindingContext()
                 throw new ArgumentNullException(nameof(modelName));
             }
 
+            var oldBindingSource = BindingSource;
```

Comment:
Remove unused variable. Or set it to `modelMetadata.BindingSource` and use it to shorten things in the next block.